src/Microsoft.AspNet.Http/Security/AuthenticateResult.cs

```diff
@@ -15,24 +13,21 @@ public class AuthenticationResult
         /// <summary>
         /// Create an instance of the result object
         /// </summary>
-        /// <param name="identity">Assigned to Identity. May be null.</param>
+        /// <param name="principal">Assigned to Principal. May be null.</param>
         /// <param name="properties">Assigned to Properties. Contains extra information carried along with the identity.</param>
         /// <param name="description">Assigned to Description. Contains information describing the authentication provider.</param>
-        public AuthenticationResult(IIdentity identity, [NotNull] AuthenticationProperties properties, [NotNull] AuthenticationDescription description)
+        public AuthenticationResult(ClaimsPrincipal principal, [NotNull] AuthenticationProperties properties, [NotNull] AuthenticationDescription description)
```

Comment:
Does this need to include the AuthenticationScheme so you can cross-reference  when requesting multiples?